pkgs/by-name/fl/flameshot/package.nix

```diff
@@ -6,55 +6,51 @@
   cmake,
   imagemagick,
   libicns,
-  libsForQt5,
+  kdePackages,
   grim,
   makeBinaryWrapper,
+  kdsingleapplication,
   nix-update-script,
-  enableWlrSupport ? false,
   enableMonochromeIcon ? false,
 }:
 
-assert stdenv.hostPlatform.isDarwin -> (!enableWlrSupport);
-
-stdenv.mkDerivation {
+stdenv.mkDerivation (finalAttrs: {
   pname = "flameshot";
-  # wlr screenshotting is currently only available on unstable version (>12.1.0)
-  version = "12.1.0-unstable-2025-05-04";
+  version = "13.0.0";
 
   src = fetchFromGitHub {
     owner = "flameshot-org";
     repo = "flameshot";
-    rev = "f4cde19c63473f8fadd448ad2056c22f0f847f34";
-    hash = "sha256-B/piB8hcZR11vnzvue/1eR+SFviTSGJoek1w4abqsek=";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-famx633wdeFVtQCg5L3JsdMLBdowYTEWNRD6hd+pMhw=";
   };
 
-  patches = [
-    # https://github.com/flameshot-org/flameshot/pull/3166
-    # fixes fractional scaling calculations on wayland
-    (fetchpatch {
-      name = "10-fix-wayland.patch";
-      url = "https://github.com/flameshot-org/flameshot/commit/5fea9144501f7024344d6f29c480b000b2dcd5a6.patch";
-      hash = "sha256-SnjVbFMDKD070vR4vGYrwLw6scZAFaQA4b+MbI+0W9E=";
-    })
-  ];
-
   cmakeFlags = [
+    "-DCMAKE_CXX_FLAGS=-I${kdsingleapplication}/include/kdsingleapplication-qt6"
+    "-DCMAKE_EXE_LINKER_FLAGS=-L${kdsingleapplication}/lib"
+    (lib.cmakeFeature "KDSingleApplication_DIR" "${kdsingleapplication}/lib/cmake/KDSingleApplication-qt6")
     (lib.cmakeBool "DISABLE_UPDATE_CHECKER" true)
     (lib.cmakeBool "USE_MONOCHROME_ICON" enableMonochromeIcon)
   ]
   ++ lib.optionals stdenv.hostPlatform.isLinux [
     (lib.cmakeBool "USE_WAYLAND_CLIPBOARD" true)
-    (lib.cmakeBool "USE_WAYLAND_GRIM" enableWlrSupport)
   ]
   ++ lib.optionals stdenv.hostPlatform.isDarwin [
-    (lib.cmakeFeature "Qt5_DIR" "${libsForQt5.qtbase.dev}/lib/cmake/Qt5")
+    (lib.cmakeFeature "Qt6_DIR" "${kdePackages.qtbase}/lib/cmake/Qt6")
+  ];
+
+  # 1. Prevents the package from attempting to fetch libraries through the internet.
+  # 2. Package expects "kdsingleapplication", in Nixpkgs: "kdsingleapplication-qt6".
+  # 3. Removes code signing procedure for Darwin builds on v13, similar to v12.
+  patches = [
+    ./flameshot.patch
   ];
```

Comment:
It would probably be easier to rebase those in the future if they would be 3 patches 😅